### PR TITLE
Feature/autonomous

### PR DESCRIPTION
**Pull request by gourab-aiplanet**

### CODE REVIEW

File: src/openagi/planner/task_decomposer.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
 from openagi.prompts.base import BasePrompt
 from openagi.prompts.constants import CLARIFIYING_VARS
 from openagi.prompts.task_clarification import TaskClarifier
-from openagi.prompts.task_creator import MultiAgentTaskCreator, SingleAgentTaskCreator
+from openagi.prompts.task_creator import MultiAgentTaskCreator
+from openagi.prompts.task_creator import AutoTaskCreator,SingleAgentTaskCreator
```
Comment (gourab-aiplanet) [inline src/openagi/planner/task_decomposer.py:17]: In the same line we can import MultiAgentTaskCreator @tarun-aiplanet

File: src/openagi/agent.py [hunk 2]
```diff
@@ -230,6 +228,64 @@ def _force_output(
...
+        Returns:
+            str: JSON of the list of Workers that needs to be executed
+        """
+
+        workers = []
+        tools_list = get_tool_list()
+        
+        # for custom tool if added by the user
+        for action in self.actions:
+        #    if action.__name__ == "MemoryRagAction":
```
Comment (gourab-aiplanet) [inline src/openagi/agent.py:249]: Remove this comment @tarun-aiplanet

File: src/openagi/planner/task_decomposer.py (hunk 1 above)
Comment (tarun-aiplanet) [reply src/openagi/planner/task_decomposer.py]: resolved

File: src/openagi/agent.py (hunk 2 above)
Comment (tarun-aiplanet) [reply src/openagi/agent.py]: resolved

**Pull request merged**